File: src/main/java/fi/fmi/mobileweather/widgets/model/WidgetData.java
package fi.fmi.mobileweather.widgets.model;

import java.util.List;

public record WidgetData(
    List<Announcement> announcements,
    List<ForecastItem> forecast,
    WarningsRecordRoot warnings,
    List<LocationRecord> location
) {
    public WidgetData(List<Announcement> announcements, List<ForecastItem> forecast) {
        this(announcements, forecast, null, null);
    }
}

File: src/main/java/fi/fmi/mobileweather/widgets/model/WidgetData.kt
package fi.fmi.mobileweather.widgets.model

data class WidgetData(
    val announcements: List<Announcement>? = null,
    val forecast: List<ForecastItem>? = null,
    val warnings: WarningsRecordRoot? = null,
    val location: List<LocationRecord>? = null
) {
    constructor(announcements: List<Announcement>?, forecast: List<ForecastItem>?) : this(
        announcements = announcements,
        forecast = forecast,
        warnings = null,
        location = null
    )
}
